### PR TITLE
feat: update ckb-ics and test sudt-transfer

### PR DESCRIPTION
**Pull request by blckngm**

### CODE REVIEW

**Review by ashuralyk — APPROVED** [review]

**Review by Flouse — APPROVED** [review]

File: tests/sudt-transfer.rs [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+            self.client_id,
+            &u16::to_be_bytes(self.channel_id)[..],
+            self.channel_contract_code_hash,
+            self.packet_contract_code_hash,
+        ]
+        .concat()
+    }
+}
+
+#[derive(Message)]
```
Comment (Flouse) [inline tests/sudt-transfer.rs:529]:
```suggestion
/// FungibleTokenPacketData defines a struct for the packet payload
/// See FungibleTokenPacketData spec:
/// <https://github.com/cosmos/ibc/tree/master/spec/app/ics-020-fungible-token-transfer#data-structures>
#[derive(Message)]
```

File: tests/sudt-transfer.rs
Comment (15168316096) [inline tests/sudt-transfer.rs:1]: This cross-chain transfer integration test does not implement verification after starting the chain node service.

**Comment by 15168316096** [comment]
The defined tokenpacketdata is somewhat inconsistent with the ics20 standard data type. Will it be compatible with the subsequent cosmos ibc and require an additional layer of data type conversion operations?
```
#[derive(Message)]
pub struct FungibleTokenPacketData {
    /// hex(sudt type script)
    #[prost(string, tag = "1")]
    pub denom: String,
    /// SUDT amount.
    #[prost(uint64, tag = "2")]
    pub amount: u64,
    /// For ckb address, this should be abi.encodePacked(ckb_blake2b(packed lock script)[..20])
    #[prost(bytes, tag = "3")]
    pub sender: Vec<u8>,
    /// For ckb address, this should be abi.encodePacked(ckb_blake2b(packed lock script)[..20])
    #[prost(bytes, tag = "4")]
    pub receiver: Vec<u8>,
}
```
ref:https://github.com/cosmos/ibc/tree/main/spec/app/ics-020-fungible-token-transfer#data-structures

File: contracts/ibc-sudt-transfer
[the remark this answers is not in the document]
Comment (blckngm) [reply contracts/ibc-sudt-transfer:1]: Good point. It's https://github.com/synapseweb3/forcerelay-ckb-contracts/pull/27. I'll add a ref next time.

File: tests/sudt-transfer.rs
Comment (blckngm) [reply tests/sudt-transfer.rs:1]: Not sure what this means. This is more like unit tests for the sudt-transfer contract. Integration test is implemented in forcerelay ibc-test.

**Comment by blckngm** [comment]
The spec is inconsistent w.r.t. json vs protobuf, and it doesn't align with existing implementations. We choose to be compatible with existing implementations, esp.

https://github.com/synapseweb3/ibc-solidity-contract/blob/master/contracts/proto/FungibleTokenPacketData.sol

File: tests/sudt-transfer.rs
Comment (15168316096) [reply tests/sudt-transfer.rs:1]: Understood, the integration test is here https://github.com/synapseweb3/forcerelay/pull/332

**Pull request merged**